### PR TITLE
Refactor ValueIndexer

### PR DESCRIPTION
**Pull request by antooni**
Allow ValueIndexer to sync multiple timestamps at once

Resolves L2B-5395

### CODE REVIEW

File: packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts [hunk 1]
```diff
@@ -11,137 +9,70 @@ import {
...
-    const prices = await this.$.priceRepo.getByTimestamp(timestamp)
-
-    const results = {
-      canonical: 0n,
-      canonicalForTotal: 0n,
-      external: 0n,
-      externalForTotal: 0n,
-      native: 0n,
-      nativeForTotal: 0n,
+    let current = from > this.$.minHeight ? from : this.$.minHeight
```
Comment (michalsidzej) [inline packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts:56]:
```suggestion
    let current = Math.max(from, this.$.minHeight)
```

File: packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts [hunk 2]
```diff
@@ -11,137 +9,70 @@ import {
...
+    while (
+      this.$.syncOptimizer.getTimestampToSync(current).toNumber() <= to &&
+      timestamps.length < this.$.maxTimestampsToProcessAtOnce
+    ) {
+      timestamps.push(this.$.syncOptimizer.getTimestampToSync(current))
+      current += 1
```
Comment (michalsidzej) [inline packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts:62]:
```suggestion
      const newTimestamp = this.$.syncOptimizer.getTimestampToSync(current)
      timestamps.push(newTimestamp)
      current = newTimestamp.toNumber() + 1
```

File: packages/backend/src/modules/tvl2/services/ValueService.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+export interface ValueServiceDependencies {
+  priceRepository: PriceRepository
+  amountRepository: AmountRepository
+}
+
+export class ValueService {
+  constructor(private readonly $: ValueServiceDependencies) {}
+
+  async getTvlAt(
```
Comment (michalsidzej) [inline packages/backend/src/modules/tvl2/services/ValueService.ts:29]: I'm not a fan of this name, as the `at` suffix was indicating that it's happening for a single timestamp.

File: packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts [hunk 4]
```diff
@@ -11,137 +9,70 @@ import {
...
-      external: 0n,
-      externalForTotal: 0n,
-      native: 0n,
-      nativeForTotal: 0n,
+    let current = from > this.$.minHeight ? from : this.$.minHeight
+    while (
+      this.$.syncOptimizer.getTimestampToSync(current).toNumber() <= to &&
+      timestamps.length < this.$.maxTimestampsToProcessAtOnce
+    ) {
+      timestamps.push(this.$.syncOptimizer.getTimestampToSync(current))
```
Comment (michalsidzej) [inline packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts:61]: Please check if there's at least one configuration to sync before adding a timestamp to this array (recent addition to `ValueIndexer`)

File: packages/backend/src/modules/tvl2/services/ValueService.ts (hunk 3 above)
Comment (antooni) [reply packages/backend/src/modules/tvl2/services/ValueService.ts:29]: yeah, good catch, it has not been renamed after abstracting it to the service

File: packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts [hunk 5]
```diff
@@ -11,137 +9,96 @@ import {
...
-
-    return timestamp.toNumber()
-  }
-
-  async getTvlAt(timestamp: UnixTime, configIds: string[]): Promise<Values> {
-    const records = await this.$.amountRepo.getByConfigIdsAndTimestamp(
-      configIds,
-      timestamp,
+    const isAnythingToCalculate = Array.from(this.amountConfigs.values()).some(
+      (config) => config.sinceTimestamp.toNumber() <= timestamps[0].toNumber(),
```
Comment (michalsidzej) [inline packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts:65]:
```suggestion
      (config) => config.sinceTimestamp.lte(timestamps[0]),
```

File: packages/backend/src/modules/tvl2/repositories/PriceRepository.ts [hunk 6]
```diff
@@ -25,9 +25,18 @@ export class PriceRepository extends BaseRepository {
     this.autoWrap<CheckConvention<PriceRepository>>(this)
   }
 
-  async getByTimestamp(timestamp: UnixTime): Promise<PriceRecord[]> {
+  async getByConfigIdsInRange(
+    configIds: string[],
+    fromInclusive: UnixTime,
```
Comment (michalsidzej) [inline packages/backend/src/modules/tvl2/repositories/PriceRepository.ts:30]: Good naming

**Review by michalsidzej — APPROVED** [review]

**Pull request merged**